### PR TITLE
636: Remove more uses of "new DocketRecord"

### PR DESCRIPTION
**Pull request by sutt0n**

### CODE REVIEW

File: shared/src/business/entities/Document.test.js [hunk 1]
```diff
@@ -1692,4 +1693,23 @@ describe('Document entity', () => {
...
+            INITIAL_DOCUMENT_TYPES.requestForPlaceOfTrial.documentType,
+          eventCode: INITIAL_DOCUMENT_TYPES.requestForPlaceOfTrial.eventCode,
+          isMinuteEntry: true,
+          isOnDocketRecord: true,
+          userId: '02323349-87fe-4d29-91fe-8dd6916d2fda',
+        },
+        { applicationContext },
+      );
+
+      expect(document.isValid()).toBe(true);
```
Comment (rachelschneiderman) [inline shared/src/business/entities/Document.test.js:1712]: Nit: should this be expecting document.isMinuteEntry to be true?

File: shared/src/business/entities/EntityConstants.js [hunk 2]
```diff
@@ -900,6 +902,10 @@ const ALL_DOCUMENT_TYPES = (() => {
     t => SYSTEM_GENERATED_DOCUMENT_TYPES[t].documentType,
   );
 
+  const minuteEntryTypes = Object.keys(MINUTE_ENTRIES_MAP)
+    .map(t => MINUTE_ENTRIES_MAP[t].documentType)
+    .filter(t => t);
```
Comment (rachelschneiderman) [inline shared/src/business/entities/EntityConstants.js:907]: What does this filter do?

Comment (sutt0n) [reply shared/src/business/entities/EntityConstants.js:907]: removes undefined entries in the array (some of the entries don't have a `documentType` property)

File: shared/src/business/useCases/updatePetitionDetailsInteractor.js [hunk 3]
```diff
@@ -63,23 +66,35 @@ exports.updatePetitionDetailsInteractor = async ({
...
       );
     } else if (isWaived) {
-      newCase.addDocketRecord(
-        new DocketRecord(
+      newCase.addDocumentWithoutDocketRecord(
+        new Document(
           {
             description: 'Filing Fee Waived',
-            eventCode: 'FEEW',
+            documentType: MINUTE_ENTRIES_MAP.filingFeeWaived.documentType,
```
Comment (rachelschneiderman) [inline shared/src/business/useCases/updatePetitionDetailsInteractor.js:90]: Love that you refactored all this stuff into EntityConstants!

**Review by rachelschneiderman — APPROVED** [review]

File: shared/src/business/entities/EntityConstants.js (hunk 2 above)
Comment (rachelschneiderman) [reply shared/src/business/entities/EntityConstants.js:907]: Thank you!

**Pull request merged**